src/main/java/DukeManager.java

```diff
@@ -0,0 +1,238 @@
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.Scanner;
+
+public class DukeManager {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________\n";
+    private static final String ADD_CONFIRMATION_MESSAGE = "Got it. I've added this task:";
+    private static final Path directoryPath = Paths.get(".","data");
+    private static final Path filePath = Paths.get(".","data", "duke.txt");
+    private static ArrayList<Task> inputs = new ArrayList<>();
+
+    /**
+     * Returns a string with formatted user inputs.
+     *
+     * @return Formatted string to print
+     */
+
+    public static String formatList() {
+        String formattedString = HORIZONTAL_LINE;
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
+                break;
+            }
+            formattedString += (i+1) + "." + inputs.get(i).toString() + "\n";
+        }
+        formattedString += HORIZONTAL_LINE;
+        return formattedString;
+    }
+
+    public static String saveList() {
+        String formattedString = "";
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
+                break;
+            }
+            formattedString += inputs.get(i).toSaveString() + "\n";
+        }
+        return formattedString;
+    }
+
+    public static void checkExists() throws IOException {
+        boolean directoryExists = Files.exists(directoryPath);
+        boolean fileExists = Files.exists(filePath);
+        if (!directoryExists) {
+            Files.createDirectories(directoryPath);
+        }
+        if (!fileExists) {
+            Files.createFile(filePath);
+        }
+    }
+
+    public static void save() {
+        try {
+            checkExists();
+        } catch (IOException e) {
+            print("Something went wrong when trying to save: " + e.getMessage());
+            return;
+        }
+        String pathName = "./data/duke.txt";
+        try {
+            FileWriter fw = new FileWriter(pathName);
+            fw.write(saveList());
+            fw.close();
+        } catch (IOException e) {
+            print("☹ OOPS!!! Something went wrong when trying to save: " + e.getMessage());
+            return;
+        }
+    }
+
+    public static void load() {
+        if (Files.exists(directoryPath) && Files.exists(filePath)) {
+            File f = new File(String.valueOf(filePath)); // create a File for the given file path
+            Scanner s = null;
+            try {
```

Comment:
consider putting all of this code in a try catch, since your error message is fairly similar and you can catch two different types of exceptions in one try catch :)